components/builder-scheduler/src/data_store.rs

```diff
@@ -259,15 +291,19 @@ impl DataStore {
         let mut rng = thread_rng();
         let id = rng.gen::<u64>();
 
-        conn.execute("SELECT insert_group_v1($1, $2)",
-                     &[&(id as i64), &project_names])
+        let (project_names, project_idents): (Vec<String>, Vec<String>) =
+            project_tuples.iter().cloned().unzip();
+
+        conn.execute("SELECT insert_group_v1($1, $2, $3)",
```

Comment:
@mwrock Per Postgres docs, the function is atomic - "Functions and trigger procedures are always executed within a transaction established by an outer query — they cannot start or commit that transaction, since there would be no context for them to execute in"